pkg/api/v1beta1/types.go

```diff
@@ -107,6 +107,15 @@ type VolumeSource struct {
 	Secret *SecretVolumeSource `json:"secret" description:"secret to populate volume with"`
 }
 
+// used by VolumeSources to describe their mounting/access modes
+type AccessModeType string
```

Comment:
Yes, these are needed by Persistent Volumes.  This PR is a small/focused one to introduce the new volume interface method suggested by @thockin to determine AccessModeType for a volume.